app/models/earning.rb

```diff
@@ -1,44 +1,27 @@
-require 'csv'
+# frozen_string_literal: true
+
 class Earning < ApplicationRecord
+  attr_accessor :external_ref
+
   belongs_to :employer
   belongs_to :employee
 
- # before_validations :save_earnings
-
-  def save_earnings(file)
-  	
-  	dates_earning = {}
-  	data = CSV.read(file)
-  	data.each_with_index  do |row, i|
-  		next if i ==0
-  		dates_earning= {external_ref: row[find_employee_id], earning_date: row[find_date_earning], earning_amount: row[find_amout]}
-  	end
-  	employee_info = get_employee(dates_earning[:external_ref])
-  	self.earning_date = format_date(dates_earning[:earning_date])
-  	self.eamount = dates_earning[:earning_amount]
-  end
+  before_validation :find_employee_id
 
-   
-  
-  def self.get_employee(external_ref)
-  	Employee.find_by(external_ref: external_ref)
+  def load_earnings(file)
```

Comment:
I don't think this method is needed. If we change `EmployerCsvService` to pass in the employer, it'll remove some complexity.

```ruby
class EmployerCsvService
  def initialize(employer:, file:)
    @employer = employer
    @file = file

    @employer_id = employer.id # or you can just directly access it when you need to get employer_id
    @csv_fields = EmployerCsvLayout.where(employer_id: employer_id) # or we can add an association on employer `has_many :employer_csv_layout` and then change this to `employer.employer_csv_layout`
  end

  # ... the rest of the mapping code
end
```

If these changes are made, you can remove this method, and change the employer method to:
```ruby
def create_earnings(file)
  EmployerCsvService.new(employer: self, file: file).process_csv
end
```

Reply:
I agree, seems a better approach to removing complexity and dependency.